modules/gapi/src/backends/oak/goakbackend.cpp

```diff
@@ -158,18 +165,25 @@ void cv::gimpl::GOAKExecutable::run(GIslandExecutable::IInput  &in,
     // FIXME: cover all outputs
     auto out_arg = out.get(0);
 
-    *cv::util::get<cv::MediaFrame*>(out_arg) = cv::MediaFrame::Create<cv::gapi::oak::OAKMediaAdapter>();
-    auto frame = cv::util::get<MediaFrame*>(out_arg);
-    auto adapter = frame->get<cv::gapi::oak::OAKMediaAdapter>();
-    adapter->setParams(cv::Size(static_cast<int>(packet->getWidth()),
-                                static_cast<int>(packet->getHeight())),
-                        cv::gapi::oak::OAKFrameFormat::BGR,
-                        packet->getData().data(),
-                        packet->getData().size());
-
-    // FIXME: do we need to pass meta here?
-    out.meta(out_arg, {});
-    out.post(std::move(out_arg));
+    // Encoder case
+    if (util::holds_alternative<cv::detail::VectorRef>(out_arg)) {
+        /*
+        *cv::util::get<cv::MediaFrame*>(out_arg) = cv::MediaFrame::Create<cv::gapi::oak::OAKMediaAdapter>();
+        auto frame = cv::util::get<MediaFrame*>(out_arg);
+        auto adapter = frame->get<cv::gapi::oak::OAKMediaAdapter>();
+        adapter->setParams(cv::Size(static_cast<int>(packet->getWidth()),
+                                    static_cast<int>(packet->getHeight())),
+                            cv::gapi::oak::OAKFrameFormat::BGR,
+                            packet->getData().data(),
```

Comment:
looks like we need consume packet here and increase it's reference counting, or simply wrap it into std::shared_ptr - something like that
to avoid dangling objects